### PR TITLE
publication statement transformation

### PR DESCRIPTION
**Pull request by reropag**
* implement publication statement transformation
* add dojson overdo class with blob
* add functions to manipulate marc21 blob

Co-Authored-by: Gianni Pante gianni.pante@rero.ch
Co-Authored-by: Peter Weber peter.weber@rero.ch

### CODE REVIEW

File: rero_ils/dojson/utils.py [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Dojson utils."""
+
+import click
+from dojson import Overdo, utils
+
+
+class ReroIlsMarc21Overdo(Overdo):
+    """Specialzed Overdo."""
```
Comment (BadrAly) [inline rero_ils/dojson/utils.py]: Specialized

File: rero_ils/dojson/utils.py [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+    lang_from_008 = None
+    date1_from_008 = None
+    date2_from_008 = None
+    date_type_from_008 = ''
+    langs_from_041_a = []
+    langs_from_041_h = []
+    alternate_graphic = {}
+
+    def __init__(self, bases=None, entry_point_group=None):
+        """Init."""
```
Comment (jma) [inline rero_ils/dojson/utils.py]: useless comment.

File: rero_ils/dojson/utils.py [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Dojson utils."""
+
+import click
+from dojson import Overdo, utils
+
+
+class ReroIlsMarc21Overdo(Overdo):
+    """Specialzed Overdo."""
```
Comment (jma) [inline rero_ils/dojson/utils.py]: can you put a more explicit comment?

File: rero_ils/dojson/utils.py [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+
+    blob_record = None
+    field_008_data = ''
+    lang_from_008 = None
+    date1_from_008 = None
```
Comment (jma) [inline rero_ils/dojson/utils.py]: is date1  start_date or from_date?

File: rero_ils/dojson/utils.py [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+        fields_044 = self.get_fields(tag='044')
+        if fields_044:
+            field_044 = fields_044[0]
+            cantons_codes = self.get_subfields(field_044, 'c')
+            for cantons_codes in self.get_subfields(field_044, 'c'):
+                self.cantons.append(cantons_codes.split('-')[1])
+            if self.cantons:
+                self.country = 'sz'
+        else:
+            self.cantons = []
```
Comment (jma) [inline rero_ils/dojson/utils.py]: seem useless

File: rero_ils/dojson/utils.py [hunk 6]
```diff
@@ -0,0 +1,245 @@
...
+        self.langs_from_041_a = []
+        self.langs_from_041_h = []
+        if len(self.field_008_data) > 38:
+            self.lang_from_008 = self.field_008_data[35:38]
+        fields_041 = self.get_fields(tag='041')
+        self.langs_from_041_a = init_lang_from(fields_041, code='a')
+        self.langs_from_041_h = init_lang_from(fields_041, code='h')
+
+    def init_alternate_graphic(self):
+        """Initialization of alternate graphic representation."""
```
Comment (jma) [inline rero_ils/dojson/utils.py]: can you add some details in comment?

File: tests/unit/test_dojson_utils.py [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+        <datafield tag="992" ind1=" " ind2=" ">
+            <subfield code="a">BFLA 148817</subfield>
+            <subfield code="x">ge/ulru/a/02/195</subfield>
+        </datafield>
+    </record>
+    """
+    rero_overdo = ReroIlsMarc21Overdo()
+    marc21json = create_record(marc21xml)
+    data = rero_overdo.do(marc21json)
+    # assert data.get('type') == 'book'
```
Comment (jma) [inline tests/unit/test_dojson_utils.py]: What do you test here? No assert!

Comment (reropag) [reply tests/unit/test_dojson_utils.py]: Deleted file tests/unit/test_dojson_utils.py

File: rero_ils/dojson/utils.py [hunk 8]
```diff
@@ -0,0 +1,254 @@
...
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Dojson utils."""
+
+import click
+from dojson import Overdo, utils
+
+
+class ReroIlsMarc21Overdo(Overdo):
+    """Specialzed Overdo.
```
Comment (BadrAly) [inline rero_ils/dojson/utils.py]: replace `Specialzed` by `Specialized`

File: rero_ils/dojson/utils.py [hunk 9]
```diff
@@ -0,0 +1,254 @@
...
+
+    The purpose of this class is to store the blob record in order to
+    have access to all the record fields during the Overdo processing.
+    This class provide also record field manupulation functions.
```
Comment (BadrAly) [inline rero_ils/dojson/utils.py]: `manupulation` by `manipulation`

**Pull request merged**